objectbox/lib/src/native/box.dart

```diff
@@ -95,6 +85,73 @@ class Box<T> {
     }
   }
 
+  /// Puts the given object in the box (persisting it) asynchronously.
+  ///
+  /// The returned future completes with an ID of a the object.  If this is a
+  /// new object (its ID property is 0), a new ID will be assigned to the object
+  /// argument, after the returned [Future] completes.
+  ///
+  /// Throws immediately if the async queue is full.
```

Comment:
1. Not sure about that, it's another level of error and if it throws, it actually allows you to stop inserting the data (for a while). But since it should be uncommon, it doesn't matter that much I guess, changed to future error.
2. Changed the comment